packages/lib/src/components/core/lume-chart/lume-chart.vue

```diff
@@ -192,6 +192,8 @@
         :position="tooltipPosition"
         :with-tooltip="allOptions.withTooltip !== false"
         :hovered-index="internalHoveredIndex"
+        :handle-mouse-enter="() => (isHoveringTooltip = true)"
+        :handle-mouse-leave="() => (isHoveringTooltip = false)"
```

Comment:
yes, of course! we're using it [here](https://github.com/Adyen/lume/pull/304/files#diff-d2a41708fc41e58377c66be1e69a80046d0e54d5902ab55fa4c8154cf78ba46bR125-R126).

Whenever the Lume consumer overrides the tooltip slot and renders tooltip by themselves - then they have to intimate us when the tooltip is hovered in/out.

In this way, we can make sure `lume-chart` is aware of tooltip current status and ensure that `mouseleave` on chart container is not called when the cursor is inside the tooltip.

Reply:
What if we move the event handlers for `@tooltip-mouseenter`, `@tooltip-mouseleave` to the slot? Then as long as the child `LumeTooltip` emits the events, it doesn't matter if the slot was overritten or not. Did you try that? Because I think it's a bad interface choice to have users manually call these functions